controllers/storageclient_controller.go

```diff
@@ -206,6 +219,43 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageClientReconciler) reconcileProviderConfig(storageClientResponse *providerpb.StorageConfigResponse) error {
+
+	for _, eResource := range storageClientResponse.ExternalResource {
+		// Create the received resources, if necessary.
+		switch eResource.Kind {
+		case "ClusterResourceQuota":
+			if err := r.reconcileClusterResourceQuota(eResource); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
```

Comment:
There is no reason to separate this logic into its own function. The above is the reconcile logic for this controller. 
In addition, the name does not make sense, as we don't have provider config as a thing that needs reconciliation. It is just another source of desired state. 

Please embed this logic into the main reconcile function (reconcilePhases) either before all other static reconciliations or after all of them.